# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/new-regexp/util/vector.h
# Commit: b3b82e1cda7f
# Full Hash: b3b82e1cda7f9f2192ab9d7bf2759b4f4b2b19fa
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-12 04:04:10
# Regressor Bug: 1634135
# File Overlap Count: 1
# Description:
#   Bug 1634135: Update shim code r=mgaudet
#   
#   There are a few changes here:
#   
#   1. The code that used to be in wrapBody was moved to RegExpCompiler::PreprocessRegExp.
# ==============================================================================

diff -r 9dddfd577a3d -r b3b82e1cda7f js/src/new-regexp/util/vector.h
--- a/js/src/new-regexp/util/vector.h	Sun May 10 16:00:25 2020 +0000
+++ b/js/src/new-regexp/util/vector.h	Sun May 10 16:00:26 2020 +0000
@@ -173,6 +173,11 @@
 template <typename T, size_t kSize>
 class SmallVector {
 public:
+  SmallVector() = default;
+  SmallVector(size_t size) {
+    resize_no_init(size);
+  }
+
   inline bool empty() const { return inner_.empty(); }
   inline const T& back() const { return inner_.back(); }
   inline void pop_back() { inner_.popBack(); };
@@ -185,6 +190,11 @@
   };
   inline size_t size() const { return inner_.length(); }
   inline const T& at(size_t index) const { return inner_[index]; }
+  T* data() { return inner_.begin(); }
+
+  T& operator[](size_t index) { return inner_[index]; }
+  const T& operator[](size_t index) const { return inner_[index]; }
+
 
   void resize_no_init(size_t new_size) {
     js::AutoEnterOOMUnsafeRegion oomUnsafe;
